test/boost/schema_change_test.cc

```diff
@@ -75,6 +80,141 @@ SEASTAR_TEST_CASE(test_new_schema_with_no_structural_change_is_propagated) {
     });
 }
 
+SEASTAR_THREAD_TEST_CASE(test_inject_internal_columns) {
+    do_with_cql_env_thread([](cql_test_env& e) {
+        auto& mm = e.migration_manager().local();
+
+        // Check internal_column_set class constructor invariants.
+        {
+            testing::scoped_no_abort_on_internal_error abort_guard{};
+
+            BOOST_REQUIRE_EXCEPTION(
+                internal_column_set("test_set", {{"test_set_c1", int32_type}}, {}),
+                std::runtime_error,
+                [](const std::runtime_error& e) {
+                    return std::string_view(e.what()).starts_with("Column set prefix \"test_set\" must start with the '$' symbol");
+                });
+            BOOST_REQUIRE_EXCEPTION(
+                internal_column_set("$test_set", {}, {}),
+                std::runtime_error,
+                [](const std::runtime_error& e) {
+                    return std::string_view(e.what()).starts_with("Column set \"$test_set\" must contain at least one column");
+                });
+            BOOST_REQUIRE_EXCEPTION(
+                internal_column_set("$test_set", {{"$test_c1", int32_type}}, {}),
+                std::runtime_error,
+                [](const std::runtime_error& e) {
+                    return std::string_view(e.what()).starts_with("Column name \"$test_c1\" must start with the column set prefix \"$test_set\"");
+                });
+        }
+
+        auto check_inject_into = [&](sstring table_name) {
```

Comment:
Please consider changing some of these lambdas - especially this one - into real functions, or, alternatively, a normal loop.
It is really hard for me to understand this code like this. I didn't realize that this test has two versions, for a view and not a view, until I went back to search for this lambda.